.circleci/config.yml

```diff
@@ -369,48 +369,20 @@ jobs:
       - setup_remote_docker:
           docker_layer_caching: true
       - run: |
-          GAIAD_VERSION=''
-          if [ "${CIRCLE_BRANCH}" = "master" ]; then
-            GAIAD_VERSION="stable"
-          elif [ "${CIRCLE_BRANCH}" = "develop" ]; then
-            GAIAD_VERSION="develop"
-          fi
-          if [ -z "${GAIAD_VERSION}" ]; then
-            docker build .
-          else
-            docker build -t tendermint/gaia:$GAIAD_VERSION .
-            docker login --password-stdin -u $DOCKER_USER <<<$DOCKER_PASS
-            docker push tendermint/gaia:$GAIAD_VERSION
-          fi
-
-  docker_tagged:
-    <<: *linux_defaults
-    steps:
-      - attach_workspace:
-          at: /tmp/workspace
-      - checkout
-      - setup_remote_docker:
-          docker_layer_caching: true
-      - run: |
-          docker build -t tendermint/gaia:$CIRCLE_TAG .
-          docker login -u $DOCKER_USER -p $DOCKER_PASS
-          docker push tendermint/gaia:$CIRCLE_TAG
+          docker build -t tendermint/gaia:${CIRCLE_BRANCH} .
+          docker login --password-stdin -u $DOCKER_USER <<<$DOCKER_PASS
+          docker push tendermint/gaia:${CIRCLE_BRANCH}
 
 workflows:
   version: 2
   test-suite:
     jobs:
       - docker_image:
-          requires:
-            - setup_dependencies
-      - docker_tagged:
           filters:
-            tags:
-              only:
-                - /^v.*/
             branches:
-              ignore:
-                - /.*/
+              only:
+                - /release\/v*/
+                - master
```

Comment:
Do we ever want to push stuff from `master`?

Reply:
Good question. In theory we could use `git describe` output as image tag. Thoughts @mircea-c?

Reply:
Bump @mircea-c

Reply:
I thought here we wanted to keep the git tag & trigger that job only when the tag was created (I've been away for some time so maybe I've missed some discussion here)

Reply:
Yes, ideally it gets triggered when the tag is created. Is that what this config does?

Reply:
It was doing it with the `docker_tagged ` job with those funny filters that @greg-szabo suggested me.
This version seems to remove the trigger at tag creation, the job will now be triggered when we push on releases branches or master branch. But as far as I can see the naming of the docker image will be something like `tendermint/gaia:release/TAG` instead of `tendermint/gaia:TAG`

Reply:
W

> Good question. In theory we could use `git describe` output as image tag. Thoughts @mircea-c?

There has been no coherent conversation on what we want to tag these images with. We can use whatever we want as long as it's something that's agreed by everyone :smile:

Reply:
@alexanderbez the last conversation we had about these images (some few months back) it was decided that it should be pushed. If that's not the case anymore, we can easily exclude it.

Reply:
We had that discussion a couple of months ago, that trigger was added to have a reference that once created won't change.
The outcome was to keep the same tag as in git so clients can reference to it.
As far as I remember we were talking about it with @jbibla @faboweb @jackzampolin and @fedekunze